test/integration/targets/openssl_privatekey/tests/validate.yml

```diff
@@ -25,4 +25,44 @@
 - name: Validate privatekey3 (assert)
   assert:
     that:
-      - privatekey1.stdout == '4096'
+      - privatekey3.stdout == '4096'
+
+
+- name: Validate privatekey4 (test)
+  stat:
+    path: '{{ output_dir }}/privatekey4.pem'
+  register: privatekey4
+
+- name: Validate privatekey4 (assert)
+  assert:
+    that:
+      - privatekey4.stat.exists == False
+
+
+- name: Validate privatekey5 (test)
+  shell: "openssl rsa -noout -text -in {{ output_dir }}/privatekey5.pem -passin pass:ansible  | grep Private | sed 's/Private-Key: (\\(.*\\) bit)/\\1/'"
+  register: privatekey5
+  when: not ansible_os_family == 'Darwin'
+
+- name: Validate privatekey5 (assert)
+  assert:
+    that:
+      - privatekey5.stdout == '4096'
+  when: not ansible_os_family == 'Darwin'
+
+- name: Validate privatekey5 idempotence (assert)
+  assert:
+    that:
+      - not privatekey5_idempotence|changed
+
+
+- name: Validate privatekey6 (test)
```

Comment:
Some of the task names, such as this one, could use a more descriptive name. Numbering them is nice for matching up failures to the task responsible, but less so for understanding the intent behind the task.